sdk/scheduler/src/main/java/com/mesosphere/sdk/scheduler/DefaultScheduler.java

```diff
@@ -555,21 +574,8 @@ private void initializeGlobals(SchedulerDriver driver) {
      * Override this function to inject your own deployment plan manager.
      */
     protected void initializeDeploymentPlanManager() {
-        LOGGER.info("Initializing deployment plan...");
-        Optional<Plan> deploy = plans.stream()
-                .filter(plan -> plan.isDeployPlan())
-                .findFirst();
-        Plan deployPlan;
-        if (!deploy.isPresent()) {
-            LOGGER.info("No deploy plan provided. Generating one");
-            deployPlan =
-                    new DeployPlanFactory(
-                            new DefaultPhaseFactory(
-                                    new DefaultStepFactory(configStore, stateStore))).getPlan(serviceSpec);
-        } else {
-            deployPlan = deploy.get();
-        }
-        deploymentPlanManager = new DefaultPlanManager(deployPlan);
+        LOGGER.info("Initializing deployment plan manager...");
+        deploymentPlanManager = new DefaultPlanManager(getDeployPlan());
```

Comment:
nitpick: could just call `getDeployPlan(plans).get()` directly, looks like this is the only place using `getDeployPlan()`